sdk/lib/atl/atlwin.h

```diff
@@ -1276,20 +1279,180 @@ class CWindow
 
 __declspec(selectany) RECT CWindow::rcDefault = { CW_USEDEFAULT, CW_USEDEFAULT, 0, 0 };
 
-template <class TBase = CWindow, class TWinTraits = CControlWinTraits>
-class CWindowImplBaseT : public TBase, public CMessageMap
+template <class TBase = CWindow>
+class CWindowImplRoot : public TBase, public CMessageMap
 {
 public:
     enum { WINSTATE_DESTROYED = 0x00000001 };
-    DWORD m_dwState;
-    const _ATL_MSG *m_pCurrentMsg;
+
+public:
     CWndProcThunk m_thunk;
+    const _ATL_MSG *m_pCurrentMsg;
+    DWORD m_dwState;
+
+    CWindowImplRoot()
+        : m_pCurrentMsg(NULL)
+        , m_dwState(0)
+    {
+    }
+
+    virtual ~CWindowImplRoot()
+    {
+    }
+};
+
+
+template <class TBase = CWindow>
+class CDialogImplBaseT : public CWindowImplRoot<TBase>
+{
+public:
+    // + Hacks for gcc
+    using CWindowImplRoot<TBase>::WINSTATE_DESTROYED;
```

Comment:
This is caused by WINSTATE_DESTROYED being scoped unnamed enum and you use it as a global enum.Maybe it's better to name this enum: `enum WinState { ... };` and do ` using CWindowImplRoot<TBase>::WinState`?

Reply:
No, it's defined in the base class, but gcc fails to resolve this due to the usage of templates.

Reply:
Well, you resolved it with `CWindowImplRoot<TBase>` part.

Reply:
It's not really GCC's fault, it's MSVC unqualified lookup extension/feature that allows it to compile. It is not part of the C++ Standard. 
You can also try `this->WINSTATE_DESTROYED`. It should work too.